x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/CrossClusterAccessAuthenticationService.java

```diff
@@ -53,21 +53,21 @@ public CrossClusterAccessAuthenticationService(
     }
 
     public void authenticate(final String action, final TransportRequest request, final ActionListener<Authentication> listener) {
-        final Authenticator.Context authcContext = authenticationService.newContext(action, request, false);
-        final ThreadContext threadContext = authcContext.getThreadContext();
-
+        final ThreadContext threadContext = clusterService.threadPool().getThreadContext();
         final CrossClusterAccessHeaders crossClusterAccessHeaders;
+        final ApiKeyService.ApiKeyCredentials apiKeyCredentials;
         try {
             // parse and add as authentication token as early as possible so that failure events in audit log include API key ID
             crossClusterAccessHeaders = CrossClusterAccessHeaders.readFromContext(threadContext);
-            final ApiKeyService.ApiKeyCredentials apiKeyCredentials = crossClusterAccessHeaders.credentials();
+            apiKeyCredentials = crossClusterAccessHeaders.credentials();
             assert ApiKey.Type.CROSS_CLUSTER == apiKeyCredentials.getExpectedType();
-            authcContext.addAuthenticationToken(apiKeyCredentials);
             apiKeyService.ensureEnabled();
         } catch (Exception ex) {
-            withRequestProcessingFailure(authcContext, ex, listener);
+            withRequestProcessingFailure(authenticationService.newContext(action, request, null), ex, listener);
```

Comment:
Passing `null` here is not very elegant, but changing this is a bigger change.

Reply:
Maybe i don't fully understand why this change is needed ( I see the 3rd boolean parameter (anon) to the newContext no longer exists) but why not continue to build the authContext early still inside the try/catch and pass in apiKeyCredential to the newContext  (i.e. call newContext instead of addAuthenticationToken) . I think as-is we will loose the api key information from the audit log/error message ?

Reply:
I think [a24b645](https://github.com/elastic/elasticsearch/pull/101384/commits/a24b6452f37e6b59fcf227d93ea16ee501c8efa5) should mitigate your concerns here?

Reply:
>I think [a24b645](https://github.com/elastic/elasticsearch/pull/101384/commits/a24b6452f37e6b59fcf227d93ea16ee501c8efa5) should mitigate your concerns here?

It does. A bit of a nitpick style preference for 1 try block v.s. 2 try blocks... but addresses my primary concern.  thanks !